### DISCUSSION TITLE
"templ generate -watch" takes up a lot of processing power

### DISCUSSION THREAD

**Discussion by nemars** [a-h/templ#302] [Q&A]
After 15 minutes bo.NextBackOff() starts returning -1 nanosecond as sleep time so the process never sleeps.

```
if firstRunComplete {
	if changesFound > 0 {
		bo.Reset()
	}
	time.Sleep(bo.NextBackOff())
}
```

Is this by expected/intended?

```
bo := backoff.NewExponentialBackOff()
bo.InitialInterval = time.Millisecond * 500
bo.MaxInterval = time.Second * 3
bo.MaxElapsedTime = 0
```

Adding MaxElapsedTime = 0 solves this since it overrides the default (15 mins) to infinity.

**Comment by a-h** [comment]
That... does not sound at all intended! I'd happily take a PR for that.

**Reply by nemars** [reply]
Sure, [here](https://github.com/a-h/templ/pull/309) it is.

Sorry for the delay, I missed the email notifying me about this reply.

**Comment by a-h** [comment]
I'll close this, because it's in the main branch. I'll do some additional testing, and make a new release soon.
